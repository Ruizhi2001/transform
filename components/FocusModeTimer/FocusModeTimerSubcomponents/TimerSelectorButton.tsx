import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import React from "react";

type Props = {
  duration: number;
  toggleTimer: (duration: number) => void;
};

export const TimerSelectorButton: React.FC<Props> = ({
  duration,
  toggleTimer,
}) => {
  return (
    <View>
      <TouchableOpacity
        style={styles.button}
        onPress={() => toggleTimer(duration)}
      >
        <Text style={styles.text}>{duration}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    padding: 10,
    margin: 10,
    borderColor: "green",
    borderWidth: 6,
    borderRadius: 10,
  },
  text: {
    color: "black",
  },
});
