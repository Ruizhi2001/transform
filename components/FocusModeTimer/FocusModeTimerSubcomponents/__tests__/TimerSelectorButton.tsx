import { TimerSelectorButton } from "../TimerSelectorButton";
import renderer from "react-test-renderer";

describe("TimerSelectorButton", () => {
  it("should match the snapshot", () => {
    const tree = renderer
      .create(<TimerSelectorButton duration={20} toggleTimer={jest.fn} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
