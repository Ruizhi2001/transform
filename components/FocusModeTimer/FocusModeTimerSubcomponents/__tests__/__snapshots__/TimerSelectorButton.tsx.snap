// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerSelectorButton should match the snapshot 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "green",
        "borderRadius": 10,
        "borderWidth": 6,
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      20
    </Text>
  </View>
</View>
`;
